azurerm/resource_arm_mssql_server_blob_auditing_policies.go

```diff
@@ -0,0 +1,230 @@
+package azurerm
+
+import (
+	"fmt"
+	"log"
+	"strings"
+	"time"
+
+	"github.com/Azure/azure-sdk-for-go/services/preview/sql/mgmt/2017-03-01-preview/sql"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/validation"
+	uuid "github.com/satori/go.uuid"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/azure"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/response"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/timeouts"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+func resourceArmMsSqlServerBlobAuditingPolicies() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceArmMsSqlServerBlobAuditingPoliciesCreateUpdate,
+		Read:   resourceArmMsSqlServerBlobAuditingPoliciesRead,
+		Update: resourceArmMsSqlServerBlobAuditingPoliciesCreateUpdate,
+		Delete: resourceArmMsSqlServerBlobAuditingPoliciesDelete,
+
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(60 * time.Minute),
+			Read:   schema.DefaultTimeout(5 * time.Minute),
+			Update: schema.DefaultTimeout(60 * time.Minute),
+			Delete: schema.DefaultTimeout(60 * time.Minute),
+		},
+
+		Schema: map[string]*schema.Schema{
+
+			"resource_group_name": azure.SchemaResourceGroupName(),
+
+			"server_name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: azure.ValidateMsSqlServerName,
+			},
+
+			"state": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validation.StringInSlice([]string{"Enabled", "Disabled"}, false),
+			},
```

Comment:
since this can be conditionally Enabled/Disabled - this wants to be nested as a block within the Server resource e.g.

```
resource "azurerm_sql_server" "test" {
  # ...
  auditing_policy {}
}
```

where we can use the presence of the `auditing_policy` block to assume Enabled/Disabled - and update this as necessary, rather than having this as a separate resource

Reply:
OK. I have made the corresponding changes in new PR https://github.com/terraform-providers/terraform-provider-azurerm/pull/5036